### DISCUSSION TITLE
How to generate static paths for index/home page?

### DISCUSSION THREAD

**Discussion by lucashfreitas** [vercel/next.js#34311] [Help] [answered]
Every single page of my website is dynamic - users can go in the backend and add pages. I implemented a catch-all pages [[...url]]. 

Users can even choose a path for a page, e.g test/test1/test2/test3 and it's working fine. My question is what we need to pass to paths to generate an index page. 

A similar question was asked a year ago in a stack overflow.

https://stackoverflow.com/questions/65587667/how-to-generate-index-html-with-getstaticpaths-in-next-js

**Comment by GabenGar** [comment]
An index page for what? For the project, just add "index.js" to the `pages` folder, since these files take precedence over catch all routes. Otherwise return different props for whatever path is considered "index" in the list.

**Comment by lucashfreitas** [comment]
@GabenGar thanks for your reply. I am aware that I can add an index.js page, but my goal is to have a generic page for every single page in the website as described, including the home page, so I would need to duplicate the logic of creating an `index.js` just because I can't create the home page using a catch-all route.

**whatever path is considered "index" in the list.**: that's my question, what path is considered index when using a catch all route?  I tried to pass `index`, ``, `/` and none of them worked. 

My question maybe is not very clear but it is how to handle an index page using a `catch-all` route, the same thing was asked over a year ago on stack overflow and did not get an answer, so maybe that's not possible?

For now, I have duplicated the catch-all routes and modified them to index, but have two identical files because is still not clear on how to do that using `getStaticPaths`.

**Reply by GabenGar** [reply]
In catch-all routes is up to you to decide the name for index pages.

**Reply by lucashfreitas** [reply]
@GabenGar don't think so. If the page has no path the static path page is not even called...

**Comment by lucashfreitas** [comment, marked as answer]
If Someone has the same question, nextjs added optional dynamic routes a while ago as described here: https://github.com/vercel/next.js/issues/10488. 

Step 1: Rename your catch all pages to `[[...slug]]``. Note that we have 2 brackets.
Step 2: On `getStaticProps`, params will be an empty object:

``` typescript

export const getStaticProps: GetStaticProps<{ data: PageData }> = async ({ params }) => {
const isHomePage = !!!params.slug // if params is an empty object and doesn't have the slug property it's the home page.
// do whatever logic will need to fetch the home page from your cms backend.

}

```

**Comment by james-wallis** [comment]
Hey @lucashfreitas, I had the same problem as you so thanks for answering your question.

On your point:
> whatever path is considered "index" in the list.: that's my question, what path is considered index when using a catch all route? I tried to pass index, ``, / and none of them worked.

While looking into the same question I found that we can pass `null`, `[]`, `undefined` or `false` to make `getStaticProps` generate the index page for a route.

> If the page uses an [optional catch-all route](https://nextjs.org/docs/routing/dynamic-routes#optional-catch-all-routes), use null, [], undefined or false to render the root-most route. For example, if you supply slug: false for pages/[[...slug]], Next.js will statically generate the page /.

(From the final point on https://nextjs.org/docs/api-reference/data-fetching/get-static-paths#paths)
